engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/EmbeddedDeploymentManager.scala

```diff
@@ -109,26 +126,9 @@ class EmbeddedDeploymentManager(modelData: ModelData, engineConfig: Config,
     (deploymentId, deploymentEntry)
   }
 
-  private def runInterpreter(processVersion: ProcessVersion, deploymentData: DeploymentData, parsedResolvedScenario: EspProcess) = {
+  private def runInterpreter(processVersion: ProcessVersion, parsedResolvedScenario: EspProcess) = {
     val jobData = JobData(parsedResolvedScenario.metaData, processVersion)
-    // TODO think about some better strategy for determining tasksCount instead of picking just parallelism for that
-    val liteKafkaJobData = LiteKafkaJobData(tasksCount = parsedResolvedScenario.metaData.typeSpecificData.asInstanceOf[LiteStreamMetaData].parallelism.getOrElse(1))
-    val interpreterTry = Try(KafkaTransactionalScenarioInterpreter(parsedResolvedScenario, jobData, liteKafkaJobData, modelData, contextPreparer))
-    interpreterTry.flatMap { interpreter =>
-      val runTry = Try {
-        val result = interpreter.run()
-        result.onComplete {
-          case Failure(exception) => handleUnexpectedError(processVersion, exception)
-          case Success(_) => //closed without problems
-        }
-      }
-      runTry.transform(
-        _ => Success(interpreter),
-        ex => {
-          interpreter.close()
-          Failure(ex)
-        })
-    }
+    deploymentStrategy.onScenarioAdded(jobData, modelData, parsedResolvedScenario, contextPreparer)
```

Comment:
wdyt about moving contextPreparer to open? this parameter is never changed so why it is passed all the time?

Reply:
:+1: